File: nc_150/trees/go/invert_bintree.go
// Just keep traversing the left and right nodes and
// swap the nodes as you go forward
// TC = O(n)
// SC = O(1)

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return root
	}

	invertTree(root.Left)
	invertTree(root.Right)

	tmp := root.Left
	root.Left = root.Right
	root.Right = tmp

	return root
}

File: nc_150/trees/invert_bin_tree.cpp
// just swap the nodes as you keep traversing towards the left and right
// TC = O(n)

TreeNode* invertTree(TreeNode* root) {
    if (!root) return root;

    invertTree(root -> left);
    invertTree(root -> right);

    TreeNode* temp = root -> left;
    root -> left = root -> right;
    root -> right = temp ;

    return root;   
}